x-pack/test/api_integration/apis/asset_manager/tests/assets.ts

```diff
@@ -123,5 +124,137 @@ export default function ({ getService }: FtrProviderContext) {
+
+        let getResponse = await supertest.get(DIFF_ENDPOINT).expect(400);
+        expect(getResponse.body.message).to.equal(
+          '[request query.aFrom]: expected value of type [string] but got [undefined]'
+        );
+
+        getResponse = await supertest.get(DIFF_ENDPOINT).query({ aFrom: timestamp }).expect(400);
+        expect(getResponse.body.message).to.equal(
+          '[request query.aTo]: expected value of type [string] but got [undefined]'
+        );
+
+        getResponse = await supertest
+          .get(DIFF_ENDPOINT)
+          .query({ aFrom: timestamp, aTo: timestamp })
+          .expect(400);
+        expect(getResponse.body.message).to.equal(
+          '[request query.bFrom]: expected value of type [string] but got [undefined]'
+        );
+
+        getResponse = await supertest
+          .get(DIFF_ENDPOINT)
+          .query({ aFrom: timestamp, aTo: timestamp, bFrom: timestamp })
+          .expect(400);
+        expect(getResponse.body.message).to.equal(
+          '[request query.bTo]: expected value of type [string] but got [undefined]'
+        );
+
+        await supertest
+          .get(DIFF_ENDPOINT)
+          .query({ aFrom: timestamp, aTo: timestamp, bFrom: timestamp, bTo: timestamp })
+          .expect(200);
+      });
+
+      it('should reject requests where either time range is moving backwards in time', async () => {
+        const now = new Date();
+        const isoNow = now.toISOString();
+        const oneHourAgo = new Date(now.getTime() - 1000 * 60 * 60 * 1).toISOString();
+
+        let getResponse = await supertest
+          .get(DIFF_ENDPOINT)
+          .query({
+            aFrom: isoNow,
+            aTo: oneHourAgo,
+            bFrom: isoNow,
+            bTo: isoNow,
+          })
+          .expect(400);
+        expect(getResponse.body.message).to.equal(
+          `Time range cannot move backwards in time. "aTo" (${oneHourAgo}) is before "aFrom" (${isoNow}).`
+        );
+
+        getResponse = await supertest
+          .get(DIFF_ENDPOINT)
+          .query({
+            aFrom: isoNow,
+            aTo: isoNow,
+            bFrom: isoNow,
+            bTo: oneHourAgo,
+          })
+          .expect(400);
+        expect(getResponse.body.message).to.equal(
+          `Time range cannot move backwards in time. "bTo" (${oneHourAgo}) is before "bFrom" (${isoNow}).`
+        );
+
+        await supertest
+          .get(DIFF_ENDPOINT)
+          .query({
+            aFrom: oneHourAgo,
+            aTo: isoNow,
+            bFrom: oneHourAgo,
+            bTo: isoNow,
+          })
+          .expect(200);
+      });
+
+      it('should return the difference in assets present between two time ranges', async () => {
+        const onlyInA = sampleAssetDocs.slice(0, 2);
+        const onlyInB = sampleAssetDocs.slice(sampleAssetDocs.length - 2);
+        const inBoth = sampleAssetDocs.slice(2, sampleAssetDocs.length - 2);
+        const now = new Date();
```

Comment:
nit: I think we can have the dates use defined once and reuse them in different tests